integrationtests/src/test/java/io/kroxylicious/proxy/ProxyRpcTest.java

```diff
@@ -72,10 +72,10 @@ public static void afterAll() {
     @MethodSource("scenarios")
     @ParameterizedTest
     public void testKroxyliciousCanDecodeManipulateAndProxyRPC(Scenario scenario) {
-        mockTester.setMockResponse(scenario.givenMockResponse());
+        mockTester.addMockResponseForApiKey(scenario.givenMockResponse());
         try (KafkaClient kafkaClient = mockTester.singleRequestClient()) {
             Response response = kafkaClient.getSync(scenario.whenSendRequest());
-            assertEquals(scenario.thenMockReceivesRequest(), mockTester.onlyRequest(), "unexpected request received at mock for scenario: " + scenario.name());
+            assertEquals(scenario.thenMockReceivesRequest(), mockTester.getOnlyRequest(), "unexpected request received at mock for scenario: " + scenario.name());
```

Comment:
I think this should be "reversed" rather than asking the `mockTester` to give you its requests maybe it would be better todo `mockTester.assertRequests(scenario. thenMockReceivesRequest())`

I'm happy to go experiment with that once this is merged.